Heimdallr/iOSHeimdallr.swift

```diff
@@ -0,0 +1,129 @@
+                                   completion: @escaping AccessTokenCompletion) {
+
+        requestAuthorizationCode(authorizationCodeURL: url,
+                                 redirectURI: redirectURI,
+                                 scope: scope,
+                                 parameters: parameters) { [weak self] result in
+
+            guard let strongSelf = self else {
+                completion(.failure(NSError(domain: "", code: 0, userInfo: nil)))
+                return
+            }
+
+            switch result {
+            case let .success(code):
+                strongSelf.requestAccessToken(authorizationCode: code, redirectURI: redirectURI, completion: completion)
+            case let .failure(error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    /// Requests an access token with the Implicit grant.
+    ///
+    /// - Parameters:
+    ///   - authorizationCodeURL: The authorization code link.
+    ///   - redirectURI: The redirect URI.
+    ///   - scope: The authorization scope.
+    ///   - completion: A cllback to invoke when the request completed.
+    public func requestAccessToken(implicitAuthorizationURL url: URL,
+                                   redirectURI: String,
+                                   scope: String,
+                                   completion: @escaping AccessTokenCompletion) {
+        requestAccessToken(implicitAuthorizationURL: url,
+                           redirectURI: redirectURI,
+                           scope: scope,
+                           responseType: "token",
+                           completion: completion)
+    }
+
+    /// Requests an authorization code.
+    ///
+    /// - Parameters:
+    ///   - authorizationCodeURL: The authorization code link.
+    ///   - redirectURI: The redirect URI.
+    ///   - scope: The authorization scope.
+    ///   - completion: A cllback to invoke when the request completed.
+    public func requestAuthorizationCode(authorizationCodeURL url: URL,
+                                         redirectURI: String,
+                                         scope: String,
+                                         parameters: [String: String],
+                                         completion: @escaping AuthorizationCodeCompletion) {
+        requestAuthorizationCode(authorizationCodeURL: url,
+                                 redirectURI: redirectURI,
+                                 scope: scope,
+                                 responseType: "code",
+                                 parameters: parameters,
+                                 completion: completion)
+    }
+
+    /// Handler for the redirect URI
+    /// Needed for requesting authorization code without `SFAuthenticationSession`.
+    /// `SFAuthenticationSession` is available since iOS 11.0
+    ///
+    /// - Parameter url: The redirect URI.
+    public func appOpen(authorizationCodeURL url: URL) {
```

Comment:
I would suggest to name this a bit more explicitly. I saw that [OAuthSwift](https://github.com/OAuthSwift/OAuthSwift) uses `handleOpenURL` for example.